# Project  2 - Problem statement

Please find below the Project for Statistical Learning course. This is an individual assignment. Kindly submit it before it's deadline.

The Titan Insurance Company has just installed a new incentive payment scheme for its lift policy sales force. It wants to have an early view of the success or failure of the new scheme. Indications are that the sales force is selling more policies, but sales always vary in an unpredictable pattern from month to month and it is not clear that the scheme has made a significant difference.

Life Insurance companies typically measure the monthly output of a salesperson as the total sum assured for the policies sold by that person during the month. For example, suppose salesperson X has, in the month, sold seven policies for which the sums assured are £1000, £2500, £3000, £5000, £10000, £35000. X's output for the month is the total of these sums assured, £61,500. Titan's new scheme is that the sales force receives low regular salaries but are paid large bonuses related to their output (i.e. to the total sum assured of policies sold by them). The scheme is expensive for the company, but they are looking for sales increases which more than compensate. The agreement with the sales force is that if the scheme does not at least break even for the company, it will be abandoned after six months.

The scheme has now been in operation for four months. It has settled down after fluctuations in the first two months due to the changeover.

To test the effectiveness of the scheme, Titan have taken a random sample of 30 salespeople measured their output in the penultimate month prior to changeover and then measured it in the fourth month after the changeover (they have deliberately chosen months not too close to the changeover). The outputs of the salespeople are shown in Table 1

 


<p> </p>
<h1>Questions</h1>
<ol>
<li>Find the mean of old scheme and new scheme column. (5 points)</li>
<li>Use the five percent significance test over the data to determine the p value to check new scheme has significantly raised outputs? (10 points)</li>
<li>What conclusion does the test (p-value) lead to? (2.5 points)</li>
<li>Suppose it has been calculated that in order for Titan to break even, the average output must increase by £5000 in the scheme compared to the old scheme. If this figure is alternative hypothesis, what is:</li>
</ol>
<p>        a) The probability of a type 1 error? (2.5 points)</p>
<p>        b) What is the p- value of the hypothesis test if we test for a difference of $5000? (10 points)</p>
<p>        c) Power of the test (5 points)</p>
<p> </p>
<p>Github link: <span> </span><a class="external" title="" href="https://github.com/GreatLearningAIML1/delhi-may19-batch-subhavkataria" target="_blank"><span><span>https://github.com/GreatLearningAIML1/delhi-may19-batch-subhavkataria</span></span></a></p>


In [64]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
%matplotlib inline
plt.style.use('ggplot')
import seaborn as sns
from scipy.stats import ttest_1samp, ttest_ind,mannwhitneyu,levene,shapiro,wilcoxon
from statsmodels.stats.power import ttest_power
from scipy import stats 

## Answer1 

In [65]:
sales_person= pd.Series([1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23,24,25,26,27,28,29,30])
old_scheme = pd.Series([57,103,59,75,84,73,35,110,44,82,67,64,78,53,41,39,80,87,73,65,28,62,49,84,63,77,67,101,91,50],name='old_scheme')
new_scheme = pd.Series([62,122,54,82,84,86,32,104,38,107,84,85,99,39,34,58,73,53,66,78,41,71,38,95,81,58,75,94,100,68],name='new_scheme')

In [66]:
old_scheme = old_scheme*1000
new_scheme = new_scheme*1000

In [67]:
scheme_df = pd.concat([old_scheme,new_scheme],axis=1)
scheme_df

old_scheme  new_scheme
0        57000       62000
1       103000      122000
2        59000       54000
3        75000       82000
4        84000       84000
5        73000       86000
6        35000       32000
7       110000      104000
8        44000       38000
9        82000      107000
10       67000       84000
11       64000       85000
12       78000       99000
13       53000       39000
14       41000       34000
15       39000       58000
16       80000       73000
17       87000       53000
18       73000       66000
19       65000       78000
20       28000       41000
21       62000       71000
22       49000       38000
23       84000       95000
24       63000       81000
25       77000       58000
26       67000       75000
27      101000       94000
28       91000      100000
29       50000       68000

In [68]:
#printing five point summary for old_scheme & new_scheme
scheme_df.describe()

old_scheme     new_scheme
count      30.000000      30.000000
mean    68033.333333   72033.333333
std     20455.980212   24062.394947
min     28000.000000   32000.000000
25%     54000.000000   55000.000000
50%     67000.000000   74000.000000
75%     81500.000000   85750.000000
max    110000.000000  122000.000000

In [8]:
old_scheme_mean = scheme_df['old_scheme'].mean()
print(" mean of old_scheme is %d thousand"%(old_scheme_mean))

 mean of old_scheme is 68033 thousand


In [11]:
new_scheme_mean = scheme_df['new_scheme'].mean()
print(" mean of new_scheme is %d thousand"%(new_scheme_mean))

 mean of new_scheme is 72033 thousand


## Answer2

### Observation from data are as folows
> -  Since population standard deviation is unkown so we have to use t stat test
-  given signifance level i.e aplha is 0.05
- sample size = 30
<br>


> #### Hypothesis Formulation
-  H0 (NULL hypothesis) : new scheme has not significantly raised outputs i.e **there is no change in means M1 == M2** 
-  H1 (Alternate hypothesis) : new scheme has significantly raised outputs i.e **there is change in means M1 < M2 or  M2 -M1 > 0**
-  Since same set of sales person sample is used and population standard deviation is unknown so we use <b>paired sample t-test</b>

In [40]:
difference_mean = int(new_scheme_mean-old_scheme_mean)
print('Mean difference is %d'%(difference_mean))
t_statistic,p_value = ttest_1samp(scheme_df['new_scheme'] - scheme_df['old_scheme'],0)
print('t_statistic is %f , p_value is %f'%(t_statistic,p_value))

Mean difference is 4000
t_statistic is 1.555914 , p_value is 0.130576


In [41]:
statistic ,pvalue =  stats.ttest_rel(scheme_df['new_scheme'], scheme_df['old_scheme'])
print(statistic ,pvalue)

1.5559143823544377 0.13057553961337662


In [47]:
p_value/2
# Since we have to check if scheme has raised outputs or not i.e we have to check on right side

0.06528776980668831

### Answer 3

>  Since p-value (=0.06528) is higher than 0.05, we **accept NULL hypothesis**. The new scheme has not significantly raised outputs.

### Answer 4

> #### Hypothesis Formulation
-  H0 : the average output must increase by £5000  i.e Mean difference Md = (M1 - M2 ) <=5000
-  H1 : the average output must increase by £5000 i.e  Mean difference Md = (M1 - M2 ) >5000
-  Since same set of sales person sample is used and population standard deviation is unknown so we use <b>paired sample t-test</b>

**Answer:**

>  ***Type I error - Rejecting Null hypothesis when it is true***
<br> Probability of Type I error is significant level or aplha, 
normally we take aplha to be 0.05 or 5%

>  
-  H0 will be rejected if t_statistic > t_critical
-  t_critical = 1.69 for 0.05 signifance level and degree of freedom is 29

In [53]:
t_critical = stats.t.ppf(1-0.05, 29)
print (t_critical)

1.6991270265334972


**What is the p- value of the hypothesis test if we test for a difference of $5000? (10 points)**

In [33]:
t_statistic,p_value = ttest_1samp(scheme_df['new_scheme'] - scheme_df['old_scheme'],5000)
print('t_statistic is %f , p_value is %f'%(t_statistic,p_value))

t_statistic is -0.388979 , p_value is 0.700133


In [82]:
standardError = ((scheme_df['new_scheme'] - scheme_df['old_scheme']).std())/np.sqrt(30)
print(standardError)
mud = 4000
xbar = t_critical* standardError
print(xbar)

2570.8355455569017
4368.176156228719


>  **NULL Hypothesis will be rejected if xbar >= 4368.176 or t_statistic >= 1.699127**
<br> **Type II error (beta) is Probability of failing to reject null when it is false. 
Power of test = (1- beta)**

In [88]:
t_statistic = (xbar- 5000 ) / standardError
t_statistic

-0.2457659514095497

In [99]:
# calculating p value for t_statistic
p_value = stats.t.cdf(t_statistic,df=29)
print('Probability of making Type II Error is %f'%(p_value))

Probability of making Type II Error is 0.403797


In [93]:
power_of_test = 1- p_value
power_of_test

0.5962027475205683

>  **Power of the test is 1 minus the probability of Type II error i.e  is 0.59 or 59%**